api/cache/cacheFactory.js

```
@@ -50,7 +60,23 @@ const violationCache = (namespace, ttl = undefined) => {
 const sessionCache = (namespace, ttl = undefined) => {
   namespace = namespace.endsWith(':') ? namespace : `${namespace}:`;
   if (!cacheConfig.USE_REDIS) return new MemoryStore({ ttl, checkPeriod: Time.ONE_DAY });
-  return new ConnectRedis({ client: ioredisClient, ttl, prefix: namespace });
+
+  try {
+    const store = new ConnectRedis({ client: ioredisClient, ttl, prefix: namespace });
+    if (ioredisClient) {
+      ioredisClient.on('error', (err) => {
+        logger.error(`Session store Redis error for namespace ${namespace}:`, err);
+      });
+    }
+
+    return store;
+  } catch (err) {
+    logger.error(
+      `Failed to create Redis session store for namespace ${namespace}, falling back to memory:`,
```

This has dangerous implications for setups that relies a shared Redis cache. If the namespace fails to establish a Redis connection, it should crash the server instead of fallback onto in-memory, since in-memory cached cannot be shared across servers and it will lead to unpredictable behaviors.

---

thanks, I'll remove the fallback